server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

```diff
@@ -254,38 +226,45 @@ public ResponseEntity<IdentityProvider> updateIdentityProvider(@PathVariable Str
 
+        return persistIdentityProviderChange(body, rawConfig, zoneId, existing, OK);
+    }
+
+    private ResponseEntity<IdentityProvider> persistIdentityProviderChange(IdentityProvider body, boolean rawConfig, String zoneId,
+            IdentityProvider existing, HttpStatus status) {
         final IdentityProvider<?> updatedIdp;
         try {
             updatedIdp = transactionTemplate.execute(txStatus -> {
-                final IdentityProvider<?> updatedOriginalIdp = identityProviderProvisioning.update(body, zoneId);
+                final IdentityProvider<?> updatedOriginalIdp = status == CREATED ? identityProviderProvisioning.create(body, zoneId) : identityProviderProvisioning.update(body, zoneId);
                 return idpAliasHandler.ensureConsistencyOfAliasEntity(updatedOriginalIdp, existing);
             });
         } catch (final IdpAlreadyExistsException e) {
             return new ResponseEntity<>(body, CONFLICT);
         } catch (final EntityAliasFailedException e) {
-            logger.warn("Could not create alias for {}", e.getMessage());
+            logger.warn(String.format("Could not create alias for %s", e.getMessage()));
             final HttpStatus responseCode = Optional.ofNullable(HttpStatus.resolve(e.getHttpStatus())).orElse(INTERNAL_SERVER_ERROR);
             return new ResponseEntity<>(body, responseCode);
         } catch (final Exception e) {
-            logger.warn("Unable to update IdentityProvider[origin=" + body.getOriginKey() + "; zone=" + body.getIdentityZoneId() + "]", e);
+            logger.warn(String.format("Unable to %s IdentityProvider[origin=%s; zone=%s]",
+                status == CREATED ? "create" : "update", body.getOriginKey(), body.getIdentityZoneId()), e);
             return new ResponseEntity<>(body, INTERNAL_SERVER_ERROR);
         }
         if (updatedIdp == null) {
-            logger.warn(
-                    "IdentityProvider[origin={}; zone={}] - Transaction updating IdP (and alias IdP, if applicable) was not successful, but no exception was thrown.",
-                    getCleanedUserControlString(body.getOriginKey()),
-                    getCleanedUserControlString(body.getIdentityZoneId())
-            );
+            if (logger.isWarnEnabled()) {
+                logger.warn(
+                    "IdentityProvider[origin={}; zone={}] - Transaction {} IdP (and alias IdP, if applicable) was not successful, but no exception was thrown.",
+                    getCleanedUserControlString(body.getOriginKey()), getCleanedUserControlString(body.getIdentityZoneId()),
+                    status == CREATED ? "creating" : "updating");
+            }
             return new ResponseEntity<>(body, UNPROCESSABLE_ENTITY);
         }
         updatedIdp.setSerializeConfigRaw(rawConfig);
         setAuthMethod(updatedIdp);
         redactSensitiveData(updatedIdp);
 
-        return new ResponseEntity<>(updatedIdp, OK);
+        return new ResponseEntity<>(updatedIdp, status);
```

Comment:
Does this not end up returning different status for the same result after this change?

Reply:
this part was a duplicated one, see
![image](https://github.com/cloudfoundry/uaa/assets/11627201/ac0cf835-a9e2-416e-8a7d-6ec0ce3691d4)

now an extra method persistIdentityProviderChange for create and update